cognite_toolkit/cdf_tk/load.py

```diff
@@ -313,120 +197,260 @@ def load_transformations(
+
+    @classmethod
+    def get_id(cls, item: T_Resource) -> T_ID:
+        raise NotImplementedError("Raw data does not have an id")
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> pd.DataFrame:
+        for pattern in [r"(\d+)\.(\w+)\.(\w+)", r"(\d+)\.(\w+)"]:
+            if match := re.match(pattern, filepath.name):
+                self.db = match.group(2) if len(match.groups()) == 3 else self.default_db
+                self.table = match.group(len(match.groups()))
+                break
+        else:
+            print(f"[bold red]ERROR:[/] Filename {filepath.name} does not match expected format.")
+            ToolGlobals.failed = True
+            return pd.DataFrame()
+        if filepath.suffix == ".csv":
+            # The replacement is used to ensure that we read exactly the same file on Windows and Linux
+            file_content = filepath.read_bytes().replace(b"\r\n", b"\n").decode("utf-8")
+            df = pd.read_csv(io.StringIO(file_content), dtype=str)
+            df.fillna("", inplace=True)
+            return df
+        elif filepath.suffix == ".parquet":
+            return pd.read_parquet(filepath)
+        else:
+            raise ValueError(f"Not supported file type {filepath.suffix}")
+
+    def create(
+        self, items: pd.DataFrame | Sequence[pd.DataFrame], ToolGlobals: CDFToolConfig, drop: bool
+    ) -> pd.DataFrame | list[pd.DataFrame] | None:
+        # This call in unnecessary, as we have the ensure_parent argument below.
+        # We keep it here just to ensure we are not doing any changes in the refactoring of the code to generalized loaders.
+        self.client.raw.databases.create(self.db)
+
+        for item in items:
+            self.client.raw.rows.insert_dataframe(
+                db_name=self.db,
+                table_name=self.table,
+                dataframe=item,
+                ensure_parent=True,
+            )
+        # Set the default db back
+        self.db = self.default_db
+        self.table = ""
+        return None
+
+
+@final
+class FileLoader(Loader[str, Path, list[Path]]):
+    load_files_individually = True
+    filetypes = frozenset()
+    api_name = "files"
+    folder_name = "files"
+    resource_cls = Path
+    id_prefix: str = "example"
+
+    @classmethod
+    def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
+        return FilesAcl(
+            [FilesAcl.Action.Read, FilesAcl.Action.Write], FilesAcl.Scope.DataSet([ToolGlobals.data_set_id])
+        )
+
+    @classmethod
+    def get_id(cls, item: Path) -> str:
+        return f"{cls.id_prefix}_{item.name}"
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> Path:
+        return filepath
+
+    def create(
+        self, items: Path | Sequence[Path], ToolGlobals: CDFToolConfig, drop: bool
+    ) -> T_Resource | T_ResourceList | None:
+        for item in items:
+            self.client.files.upload(
```

Comment:
One could argue that we should add support for explicitly declaring files as well, so that asset mapping can be done. That's more of a feature request though, but maybe worth capturing as a TODO or jira?

Reply:
Lets use Jira, do not like todos in code (as I think they should instead just be fixed immediately), and this sounds like a feature.